### PR TITLE
feat(ct): add SOLIDITY_UPGRADES document

### PR DESCRIPTION
**Pull request by smartcontracts**
Adds SOLIDITY_UPGRADES.md to contracts-bedrock. Please note that this document makes some references to a unified Solidity version. This will be true in a few days...

### CODE REVIEW

File: packages/contracts-bedrock/README.md [hunk 1]
```diff
@@ -71,6 +71,12 @@ Please refer to the [CONTRIBUTING.md](https://github.com/ethereum-optimism/optim
 OP Stack smart contracts should be written according to the [STYLE_GUIDE.md](./STYLE_GUIDE.md) found within this repository.
 Maintaining a consistent code style makes code easier to review and maintain, ultimately making the development process safer.
 
+### Solidity Versioning
```
Comment (mds1) [inline packages/contracts-bedrock/README.md:75]: need to add this to the table of contents

File: packages/contracts-bedrock/SOLIDITY_UPGRADES.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+**Important**: New Solidity versions must not be introduced to any part of the codebase without
+going through the formal version update proposal process outlined in this document.
+
+## Update Process
+
+1. **Minimum Delay Period**: A new Solidity version must be at least 6 months old before it can be
+  considered for adoption.
+2. **Proposal Submission**: Before any Solidity version upgrades are made, a detailed proposal must
+  be submitted as a pull request to the `ethereum-optimism/design-docs` repository in the
```
Comment (mds1) [inline packages/contracts-bedrock/SOLIDITY_UPGRADES.md]:
```suggestion
  be submitted as a pull request to the [`ethereum-optimism/design-docs`](https://github.com/ethereum-optimism/design-docs/) repository in the
```

File: packages/contracts-bedrock/SOLIDITY_UPGRADES.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  `solidity/` subfolder, following the standardized format outlined below. This applies to the
```
Comment (mds1) [inline packages/contracts-bedrock/SOLIDITY_UPGRADES.md:21]: In the design docs repo's `assets` folder we have an FMA template. I'd suggest moving this template there as well

File: packages/contracts-bedrock/SOLIDITY_UPGRADES.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+[Explain specific feature(s) in the new version that would benefit the OP Stack codebase, with a clear explanation of their potential impact and usefulness]
+
+## Summary of Changes
+
+[Provide a comprehensive summary of all significant features, changes, and bug fixes introduced between the current version and the proposed upgrade]
+
+## High-Risk Features or Notable Bug Fixes
+
+[Highlight any high-risk features or notable bug fixes that require special attention during the review process]
```
Comment (mds1) [inline packages/contracts-bedrock/SOLIDITY_UPGRADES.md]: What is a high risk feature?

Comment (smartcontracts) [reply packages/contracts-bedrock/SOLIDITY_UPGRADES.md]: Yeah good point, I'm going to make that separate PR into design-docs first then we can get back to this

Comment (smartcontracts) [reply packages/contracts-bedrock/SOLIDITY_UPGRADES.md]: https://github.com/ethereum-optimism/design-docs/pull/104

File: packages/contracts-bedrock/SOLIDITY_UPGRADES.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+The review panel will evaluate each proposal based on the criteria mentioned in the "Review and Approval" section above. They may request additional information or clarifications if needed.
+
+## Implementation
+
+If approved, the Solidity version upgrade will be implemented across the entire OP Stack codebase. This process will be managed by the development team to ensure consistency and minimize potential issues. The upgrade will apply to all contracts and components simultaneously.
+
+<!-- References -->
+[1]: https://github.com/ethereum-optimism/design-docs
+[2]: https://github.com/ethereum-optimism/design-docs/tree/main/solidity
+[3]: https://github.com/ethereum-optimism/design-docs/tree/main/assets/solc-update.md
```
Comment (mds1) [inline packages/contracts-bedrock/SOLIDITY_UPGRADES.md]: The PR calls the file `solc-update-template.md`
```suggestion
[3]: https://github.com/ethereum-optimism/design-docs/tree/main/assets/solc-update-template.md
```

**Review by mds1 — APPROVED** [review]

**Comment by yorhodes** [comment]
as a reference, we added this recently too  https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4518

**Pull request merged**